aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIOProperties.java

```diff
@@ -0,0 +1,394 @@
+    Assert.assertNull(s3FileIOProperties.s3SessionToken());
+    Assert.assertNull(s3FileIOProperties.s3FileIoAcl());
+    Assert.assertNull(s3FileIOProperties.s3Endpoint());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_PRELOAD_CLIENT_ENABLED_DEFAULT,
+        s3FileIOProperties.s3PreloadClientEnabled());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_DUALSTACK_ENABLED_DEFAULT, s3FileIOProperties.isS3DualStackEnabled());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3FILEIO_PATH_STYLE_ACCESS_DEFAULT,
+        s3FileIOProperties.isS3PathStyleAccess());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_USE_ARN_REGION_ENABLED_DEFAULT,
+        s3FileIOProperties.isS3UseArnRegionEnabled());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_ACCELERATION_ENABLED_DEFAULT,
+        s3FileIOProperties.isS3AccelerationEnabled());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_REMOTE_SIGNING_ENABLED_DEFAULT,
+        s3FileIOProperties.isS3RemoteSigningEnabled());
+
+    Assert.assertEquals(
+        Runtime.getRuntime().availableProcessors(),
+        s3FileIOProperties.s3FileIoMultipartUploadThreads());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3FILEIO_MULTIPART_SIZE_DEFAULT,
+        s3FileIOProperties.s3FileIoMultiPartSize());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3FILEIO_MULTIPART_THRESHOLD_FACTOR_DEFAULT,
+        s3FileIOProperties.s3FileIOMultipartThresholdFactor(),
+        0.0);
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3FILEIO_DELETE_BATCH_SIZE_DEFAULT,
+        s3FileIOProperties.s3FileIoDeleteBatchSize(),
+        0.0);
+
+    Assert.assertEquals(
+        System.getProperty("java.io.tmpdir"), s3FileIOProperties.s3fileIoStagingDirectory());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_CHECKSUM_ENABLED_DEFAULT, s3FileIOProperties.isS3ChecksumEnabled());
+
+    Assert.assertEquals(Sets.newHashSet(), s3FileIOProperties.s3WriteTags());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_WRITE_TABLE_TAG_ENABLED_DEFAULT,
+        s3FileIOProperties.s3WriteTableTagEnabled());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_WRITE_NAMESPACE_TAG_ENABLED_DEFAULT,
+        s3FileIOProperties.s3WriteNamespaceTagEnabled());
+
+    Assert.assertEquals(Sets.newHashSet(), s3FileIOProperties.s3DeleteTags());
+
+    Assert.assertEquals(
+        Runtime.getRuntime().availableProcessors(), s3FileIOProperties.s3FileIoDeleteThreads());
+
+    Assert.assertEquals(
+        S3FileIOProperties.S3_DELETE_ENABLED_DEFAULT, s3FileIOProperties.isS3DeleteEnabled());
+
+    Assert.assertEquals(Collections.emptyMap(), s3FileIOProperties.s3BucketToAccessPointMapping());
+  }
+
+  @Test
+  public void testS3FileIOProperties() {
+    Map<String, String> map = getTestProperties();
+    S3FileIOProperties s3FileIOProperties = new S3FileIOProperties(map);
+
+    Assert.assertEquals(
```

Comment:
I think a slightly better alternative to testing this would be: `Assertions.assertThat(map).containsEntry(S3FileIOProperties.S3FILEIO_SSE_TYPE), s3FileIOProperties.s3FileIoSseType())`.

The main advantage here is that it will show the content of `map` in case the assertion ever fails.